pkg/controller/dr/drcontroller.go

```diff
@@ -71,8 +93,29 @@ func (d *DrController) CreateReplication(ctx *c.Context, replica *ReplicationSpe
 	replica.PrimaryReplicationDriverData = utils.MergeStringMaps(replica.PrimaryReplicationDriverData, primaryVol.Metadata)
 	replica.SecondaryReplicationDriverData = utils.MergeStringMaps(replica.SecondaryReplicationDriverData, secondaryVol.Metadata)
 
-	// Load replication operator by replication type.
-	d.LoadOperator(pPool.ReplicationType)
+	// Load replication operator
+	d.LoadOperator(ctx, primaryVol, secondaryVol)
+
+	// Host-Based replication needs to do some extra operations including attach adn volume data list
```

Comment:
"adn"?  This comment needs to be rephrased.  I don't know what you are trying to say.